app/forms/common/Eori.scala

```diff
@@ -39,6 +39,6 @@ object Eori {
   def mapping(errorPrefix: String): Mapping[Eori] =
     text()
       .verifying(s"${errorPrefix}.eori.empty", nonEmpty)
-      .verifying(s"${errorPrefix}.eori.error.format", isValidEORIPattern and noLongerThan(17) and noShorterThan(3))
+      .verifying(s"${errorPrefix}.eori.error.format", isEmpty or (isValidEORIPattern and noLongerThan(17) and noShorterThan(3)))
```

Comment:
Is this safe doing this on a "common" component?

Reply:
This component was created, but had wrong validation, because fo empty show two errors.

I don't like this approach, I like wrapping Strings in class to be more descriptive and type safe, but not sure if there is any value in this Eori class.